Examples/restcomm-olympus-nopod/restcomm-olympus/PushSettingsViewController.m

```diff
@@ -0,0 +1,288 @@
+        // in the navigation stack.
+        [self update];
+    }
+    [super viewWillDisappear:animated];
+}
+
+- (void)hideKeyBoard
+{
+    // resign active first responder if available
+    if (self.activeField) {
+        [self.activeField resignFirstResponder];
+    }
+}
+
+- (void)update
+{
+    if (self.enableSwitch.on){
+        //trim settings inputs for spaces
+        NSString * pushAccount = [self.pushAccountText.text stringByTrimmingCharactersInSet:
+                                  [NSCharacterSet whitespaceCharacterSet]];
+        
+        NSString * pushPassword = [self.pushPasswordText.text stringByTrimmingCharactersInSet:
+                                  [NSCharacterSet whitespaceCharacterSet]];
+       
+        NSString * pushDomain = [self.pushDomainText.text stringByTrimmingCharactersInSet:
+                                   [NSCharacterSet whitespaceCharacterSet]];
+       
+        if ([pushAccount length] == 0){
+            [Utils shakeTableViewCell:self.accountTVCell];
+            return;
+        }
+        
+        if ([pushPassword length] == 0){
+            [Utils shakeTableViewCell:self.passwordTVCell];
+            return;
+        }
+        
+        if ([pushDomain length] == 0){
+            [Utils shakeTableViewCell:self.domainTVCell];
+            return;
+        }
+       
+        [self showSpinner];
+        [[UIApplication sharedApplication] beginIgnoringInteractionEvents];
+        
+        //save account, password and domain
+        [Utils updatePushAccount:pushAccount];
+        [Utils updatePushPassword:pushPassword];
+        [Utils updatePushDomain:pushDomain];
+        
+        //get certificate strings
+        NSString *pushCertificatesPathPublic = [[NSBundle mainBundle] pathForResource:@"certificate_key_push" ofType:@"pem"];
+        NSString *pushCertificatesPathPrivate = [[NSBundle mainBundle] pathForResource:@"rsa_private_key_push" ofType:@"pem"];
+        
+        NSMutableDictionary *dic = [[NSMutableDictionary alloc] initWithObjectsAndKeys:
+                                    kFriendlyName, @"friendly-name",
+                                    pushAccount, @"rescomm-account-email",
+                                    pushPassword, @"password",
+                                    pushDomain, @"push-domain",
+                                    [Utils pushToken], @"token",
+                                    pushCertificatesPathPublic, @"push-certificate-public-path",
+                                    pushCertificatesPathPrivate, @"push-certificate-private-path",
+                                    [NSNumber numberWithBool:[Utils isSandbox]], @"is-sandbox", nil];
+        
+        AppDelegate *appDelegate = ((AppDelegate *)[UIApplication sharedApplication].delegate);
+        RCDevice  *rcDevice = [appDelegate registerRCDevice];
+        if (rcDevice){
+            [rcDevice registerPushToken:dic delegate:self];
```

Comment:
Is there any case where we dont need to call register push but need to register device? Currently I can see that we are registering RCDevice when push payload is received (in this case we dont need to register for push, because we are already registered)  
If we want to have it in the init, then I think we should have separate method initWithPush or something similar. Do we want this? We can talk about this more on a call.